drafts/0007-RFC-BAG-OPERATORS.md

```diff
@@ -44,279 +44,288 @@ INTERSECT(A, B) = DISTINCT(INTERSECT_ALL(A, B))
+|---------|-------|--------|----------|
+| 0       | LEVEL | Int    | NO       |
+| 1       | NAME  | String | NO       |
+| 2       | TITLE | String | NO       |
+
+        engineering.employees
+
+| Ordinal | Name  | Type   | Nullable |
+|---------|-------|--------|----------|
+| 0       | TITLE | String | NO       |
+| 1       | NAME  | String | NO       |
+```
+
+### Union Examples
+
+**Union 1** — Homogenous Bags — Both strict and permissive typing would produce this result.
+```sql
+SELECT name, title FROM hr.employees
+UNION ALL
+SELECT name, title FROM engineering.employees
+
+<<
   {
-    'id': 1,
-    'name': 'Bob Smith'
+    'name': 'Bob Smith',
+    'title': 'Consultant'
   },
   {
-    'id': 3,
-    'name': 'Jane Smith'
+    'name': 'Susan Smith',
+    'title': 'Manager'
   },
   {
-    'id': 1,
-    'name': 'Quincy Jones'
+    'name': 'Jane Smith',
+    'title': 'DEI'
   },
   {
-    'id': 3,
-    'name': 'Indiana Jones'
+    'name': 'Paul McCharmly',
+    'title': 'Manager'
   },
   {
-    'id': 4,
-    'name': 'Eric Lennon'
+    'name': 'George Parasol',
+    'title': 'SDE'
   },
   {
-    'id': 3,
-    'name': 'Ringo Stone'
+    'name': 'Ringo Stone',
+    'title': 'Principal SDE'
   },
   {
-    'id': 2,
-    'name': 'George Parasol'
+    'name': 'Eric Lennon',
+    'title': 'Intern'
   }
-]
----
-OK!
+>>
+-- OK!
 ```
 
-**Union 2** — Union where right operands are expressions rather than SFW blocks. Note that this is stated as possible in the SQL++ text, but the grammar does not support this. This is the result of an initial implementation because our parser actually parsers UNION operators as expressions rather than SFW blocks as described by the grammar.
+**Union 2** — Homogenous Bags with strict typing
 
-```
-PartiQL> SELECT name FROM (
-   |   SELECT * FROM hr.employees UNION engineering.employees UNION accounting.employees
-   | ) WHERE title = 'Intern'
-   |
-==='
-<<
-  {
-    'name': 'Eric Lennon'
-  },
-  {
-    'name': 'Indiana Jones'
-  }
->>
----
-OK!
-```
+Here is **union 1** with strict typing, but the **names** of the column descriptors do not match. Notice the lack of selection list here. These two relations are union-compatible because the i-th column of *hr.employees* is comparable to the i-th column of *engineering.employees*, but notice the column names. This behavior conforms to SQL specification whereby, if the column names do not match, then the column name is implementation-dependent i.e. it is not derived from any particular value.
 
-**Union 3** -- Get all distinct titles
-```
-PartiQL> SELECT DISTINCT title FROM (
-   |   SELECT * FROM hr.employees
-   |   UNION
-   |   SELECT * FROM accounting.employees
-   |   UNION
-   |   SELECT * FROM engineering.employees
-   | )
-   | ORDER BY title ASC
-   |
-==='
-[
+```sql
+SELECT * FROM hr.employees
+UNION ALL
+SELECT * FROM engineering.employees
+
+<<
   {
-    'title': 'Consultant'
+    '_0': 'Bob Smith',
+    '_1': 'Consultant'
```

Comment:
I am curious how Postgres/MySQL/Oracle deal with the column aliasing.  This might be one of those strictly correct cases, but practically we take the labels from one relation.